src/memory.cc

```diff
@@ -0,0 +1,15 @@
+#include "memory.h"
+
+Memory::Memory(std::vector<uint8_t> data)
+    : data_(std::move(data)) {}
+
+Memory::~Memory() = default;
+
+Memory::Status Memory::Read(uint16_t address, uint8_t* byte) {
+  *byte = data_[address];
```

Comment:
Could |byte| be a null pointer or something weird like that?

Reply:
Yeah, I'll check for nullptr and add some tests.